src/servers/http_server_v2.cc

```diff
@@ -0,0 +1,1566 @@
+    TRTSERVER_InferenceResponse* response, void* userp)
+{
+  HTTPAPIServerV2::InferRequestClass* infer_request =
+      reinterpret_cast<HTTPAPIServerV2::InferRequestClass*>(userp);
+
+  // write outputs into json array
+  int i = 0;
+  for (auto ev_buffer : infer_request->response_meta_data_.response_buffer_) {
+    size_t buffer_len = evbuffer_get_length(ev_buffer);
+    char json_buffer[buffer_len];
+    evbuffer_copyout(ev_buffer, &json_buffer, buffer_len);
+    WriteDataArrayToJson(
+        &infer_request->response_meta_data_.response_json_, i++, json_buffer,
+        buffer_len);
+  }
+
+  // write json string into buffer
+  rapidjson::StringBuffer buffer;
+  buffer.Clear();
+  rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
+  infer_request->response_meta_data_.response_json_.Accept(writer);
+  std::string infer_metadata(buffer.GetString());
+
+  evbuffer_add(
+      infer_request->req_->buffer_out, infer_metadata.c_str(),
+      infer_metadata.size());
+
+  if (infer_request->FinalizeResponse(response) == EVHTP_RES_OK) {
+    evthr_defer(infer_request->thread_, OKReplyCallback, infer_request);
+  } else {
+    evthr_defer(infer_request->thread_, BADReplyCallback, infer_request);
+  }
+
+  // Don't need to explicitly delete 'trace_manager'. It will be deleted by
+  // the TraceMetaData object in 'infer_request'.
+  LOG_TRTSERVER_ERROR(
+      TRTSERVER_InferenceResponseDelete(response), "deleting HTTP response");
+}
+
+evhtp_res
+HTTPAPIServerV2::InferRequestClass::FinalizeResponse(
+    TRTSERVER_InferenceResponse* response)
+{
+  InferResponseHeader response_header;
+
+  TRTSERVER_Error* response_status =
+      TRTSERVER_InferenceResponseStatus(response);
+  if (response_status == nullptr) {
+    TRTSERVER_Protobuf* response_protobuf = nullptr;
+    response_status =
+        TRTSERVER_InferenceResponseHeader(response, &response_protobuf);
+    if (response_status == nullptr) {
+      const char* buffer;
+      size_t byte_size;
+      response_status =
+          TRTSERVER_ProtobufSerialize(response_protobuf, &buffer, &byte_size);
+      if (response_status == nullptr) {
+        if (!response_header.ParseFromArray(buffer, byte_size)) {
+          response_status = TRTSERVER_ErrorNew(
+              TRTSERVER_ERROR_INTERNAL, "failed to parse response header");
+        }
+      }
+
+      TRTSERVER_ProtobufDelete(response_protobuf);
+    }
+  }
+
+  if (response_status == nullptr) {
+    std::string format;
+    const char* format_c_str = evhtp_kv_find(req_->uri->query, "format");
+    if (format_c_str != NULL) {
+      format = std::string(format_c_str);
+    } else {
+      format = "text";
+    }
+
+    // The description of the raw outputs needs to go in the
+    // kInferResponseHTTPHeader since it is needed to interpret the
+    // body. The entire response (including classifications) is
+    // serialized at the end of the body.
+    response_header.set_id(request_id_);
+
+    std::string rstr;
```

Comment:
There are no query parameters in the new API, and this seems to be returning protobuf response header?